tests/Valkey.Glide.TestUtils/TempFile.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+namespace Valkey.Glide.TestUtils;
+
+/// <summary>
+/// Temporary file that is deleted when disposed.
+/// </summary>
+public sealed class TempFile : IDisposable
+{
+    public string Path { get; }
+
+    public byte[] Bytes
+    {
+        get => File.ReadAllBytes(Path);
+        set => File.WriteAllBytes(Path, value);
+    }
+
+    public string Text
+    {
+        get => File.ReadAllText(Path);
+        set => File.WriteAllText(Path, value);
+    }
+
+    public string[] Lines
+    {
+        get => File.ReadAllLines(Path);
+        set => File.WriteAllLines(Path, value);
+    }
+
+    public TempFile()
+    {
+        Path = System.IO.Path.GetTempFileName();
+    }
+
+    public TempFile(byte[] contents) : this()
+    {
+        File.WriteAllBytes(Path, contents);
+    }
+
+    public TempFile(string contents) : this()
+    {
+        File.WriteAllText(Path, contents);
+    }
+
+    public void Dispose() => File.Delete(Path);
```

Comment:
We could make this class more safe by implementing the destructor to call Dispose() in case the caller forgets to Dispose().

We can also create the file with the DeleteOnExit flag.